src/internal/packager/seed.go

```diff
@@ -103,14 +103,32 @@ func preSeedRegistry(tempPath tempPaths) {
 	}
 
 	// CLI provided overrides that haven't been processed already
-	if config.DeployOptions.NodePort != "" {
-		state.NodePort = config.DeployOptions.NodePort
+	if config.InitOptions.NodePort != "" {
+		state.NodePort = config.InitOptions.NodePort
 	}
-	if config.DeployOptions.Secret != "" {
-		state.Secret = config.DeployOptions.Secret
+	if config.InitOptions.Secret != "" {
+		state.Secret = config.InitOptions.Secret
 	}
-	if config.DeployOptions.StorageClass != "" {
-		state.StorageClass = config.DeployOptions.StorageClass
+	if config.InitOptions.StorageClass != "" {
+		state.StorageClass = config.InitOptions.StorageClass
+	}
+
+	if config.InitOptions.GitServer.Address != "" {
+		// The user has provided external git server info, just use what they gave!
+		state.GitServer = config.InitOptions.GitServer
+
+		// For external clusters, the read-user is the same as the push-user
+		state.GitServer.ReadUsername = state.GitServer.PushUsername
```

Comment:
I still think we need to look at this--I don't see why wouldn't allow the user to specify a read-only user as well--it would likely be a robot account, PAT or similar to grant less-privilege and I think it would be a mistake to not allow them to do that.  Think this is related to assuming we can push to a different git server just like we do for gitea.

Reply:
Let me know when you have a bit of time to chat about this. Just want to make sure we're both on the same page :)